### PR TITLE
[REVIEW] Add Apache arrow 0.10 support

### PR DESCRIPTION
**Pull request by pearu**
This PR makes arrow dependent (via libgdf) tests independent of particular arrow version.

For libgdf with arrow 0.10 support, see https://github.com/gpuopenanalytics/libgdf/pull/89

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by kkraus14** [comment]
add to whitelist

File: pygdf/tests/test_gpu_arrow_parser.py [hunk 1]
```diff
@@ -1,13 +1,81 @@
...
+                    33.66364, 30.194532, 41.785984, 36.124477,
+                    28.428888, 41.785984, 29.533695],
+                   dtype=np.float32)
+    lon = np.array([-94.713905, -87.752426, -98.46978, -87.752426,
+                    -94.713905, -76.6682, -95.27889, -86.678185,
+                    -98.46978, -96.85177,
+                    -98.46978, -96.85177, -87.752426, -85.736,
+                    -80.23287, -87.752426,
+                    -101.82278, -97.66987, -87.752426, -86.678185, -81.316025,
+                    -87.752426, -98.46978], dtype=np.float32)
```
Comment (kkraus14) [inline pygdf/tests/test_gpu_arrow_parser.py]: Could we generate this data using a static seed instead of hardcoding it in?

**Comment by kkraus14** [comment]
@pearu It looks like a couple of the tests are failing because of the `ipums.pkl` file isn't being properly read.

**Comment by kkraus14** [comment]
Also, I'd be in favor of dropping the hardcoded Arrow 0.7.1 binary data tests and moving it all to the pyarrow generated data and adding pyarrow as a test dependency.

**Comment by pearu** [comment]
@kkraus14 , the tests are failing because pyarrow is not installed. 
Where is gpuCI/pygdf-prb environment defined? I cannot find it under pygdf repo.

I'll drop the hardcoded 0.7.1 arrow binary data.

**Comment by kkraus14** [comment]
@mike-wendt Can you address the above question?

File: pygdf/tests/test_gpu_arrow_parser.py (hunk 1 above)
Comment (pearu) [reply pygdf/tests/test_gpu_arrow_parser.py]: Sure, I'll do it.

Comment (pearu) [reply pygdf/tests/test_gpu_arrow_parser.py]: Done.

**Comment by kkraus14** [comment]
@pearu Looks good to me! @scopatz Would you mind reviewing as well?

**Comment by scopatz** [comment]
This LGTM! Thanks

**Comment by kkraus14** [comment]
Thanks @pearu! Merging.

**Pull request merged**